samples/stream_builder_custom.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * This file is part of the Camera Streaming Daemon
+ *
+ * Copyright (C) 2017  Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "stream_builder.h"
+
+class StreamBuilderCustom : public StreamBuilder {
+public:
+    StreamBuilderCustom() : StreamBuilder() {
+    }
+    ~StreamBuilderCustom() {}
```

Comment:
Using virtual for destructor is not recommended for classes that doesn't intend to be extended and that's the case for StreamBuilderCustom and StreamBuilderRealsense.
Note that the base class has a virtual method, so the base destructor will be called when StreamBuilderCustom and StreamBuilderRealsense are destroyed.